torch/onnx/_type_utils.py

```diff
@@ -108,36 +113,96 @@ def from_name(
     @classmethod
     @_beartype.beartype
     def from_dtype(cls, dtype: torch.dtype) -> JitScalarType:
-        """Convert a torch dtype to ScalarType."""
+        """Convert a torch dtype to JitScalarType.
+
+        NB: DO NOT USE this API if dtype is comes from a torch._C.Value.type().dtype() call because
+            "RuntimeError: INTERNAL ASSERT FAILED at "../aten/src/ATen/core/jit_type_base.h" can
+            be raised in some scenarios. Instead use from_value API which handles all cases
+
+        Args:
+            dtype: A torch.dtype to create a JitScalarType from
+
+        Returns:
+            JitScalarType
+
+        Raises:
+            ValueError: if dtype is not a valid torch.dtype or if it is None.
+        """
         if dtype not in _DTYPE_TO_SCALAR_TYPE:
             raise ValueError(f"Unknown dtype: {dtype}")
         return _DTYPE_TO_SCALAR_TYPE[dtype]
 
+    # TODO: `value: torch._C.Value` type annotation raises (CI machines only):
+    #       NameError: name '_C' is not defined
+    # TODO: Return type Optional[JitScalarType] or Union[JitScalarType, None] raises (CI machines only):
```

Comment:
That's strange 🤔

Reply:
I still don't understand why this would happen. Does lintrunner give the same error locally? I will test it on my machine some time too

Reply:
As the comment states, `(CI machines only)`

For some reason, `_type_utils.py` goes crazy with `torch._C.*`, `from torch import _C`. Didn't do `from torch._C import Value` as it would be well accepted on the code review

Reply:
I _think_ it is because JitScalarType is imported in the `__init__` of torch.onnx, so there is some circular import going on when torch initializes.

Reply:
Did you confirm this issue on your experiments?

Reply:
on it